src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceFileLogger.cs

```diff
@@ -87,7 +92,11 @@ public virtual async Task ProcessLogQueue(object state)
             while (!_cancellationTokenSource.IsCancellationRequested)
             {
                 await InternalProcessLogQueue();
-                await Task.Delay(TimeSpan.FromSeconds(FlushFrequencySeconds), _cancellationTokenSource.Token).ContinueWith(task => { });
+                await Task.Delay(TimeSpan.FromSeconds(_flushFrequencySeconds), _cancellationTokenSource.Token).ContinueWith(task => { });
+                if ( _currentFlushFrequencySeconds < MaxFlushFrequencySeconds)
```

Comment:
The way you have this structured now we're never resetting back to 1, so once you increment to 32 you'll stay there. I think instead you want InternalProcessLogQueue to return a bool indicating whether anything was written. Then, you only back off if there are no events written, otherwise you reset to 1 (meaning we're active again and need to be more responsive). Looks like you're not using this new _currentFlushFrequencySeconds value above? I think you'll want to set this value prior then use it in the delay.

Also, you'd want to use _currentFlushFrequencySeconds = Min(MaxFlushFrequencySeconds, _currentFlushFrequencySeconds * FlushFrequencyGrowthFactor) so we don't go over max.

In practice, you should see how far you actually end up backing off. Because all our system logs go through this logger, I'd expect even if no functions are executing, some of our other periodic logs (like lease acquisitions, etc.) would keep things hot, so the question is whether we'd end up always using 1s or a really low number in most cases.

Reply:
Thank you for your review Mathew. In this case, my goal is to emit the first FunctionExecutionCount log as fast as possible.  After the first log is emitted, I don't mind waiting 30 seconds between emissions. Platform picks up the any available FunctionExecutionCount log and reports to the DataService that a worker is performing and can be scaled. 

I have incorporated lots of your suggestions. I appreciate the thorough review. I don't think I have a complete grasp of how the File Loggers are working. My current understanding is that this 30 seconds delay is not actually a maximum. I may have that incorrect. But, in private testing I found a marked increase in scaling speed when I implemented the backoff. Therefore, I think that the loggers are only filing every `FlushFrequency` seconds. Atleast it seems that way for the very first log emission.

Reply:
If that's the case, then I guess I don't understand these changes. This logger is used for all logs, not only function execution logs. What is the "FunctionExecutionCount" log you're referring to? There's no guarantees on what logs will happen first after startup, so how do you know what ends up in the first batch to be flushed? Its unclear what you're trying to accomplish here.

The current logger will only flush logs every 30s. It accumulates them and flushes them in batches. With the changes you're making, it'll flush at after 1s, then 2s, then 4s, ... 30s, 30s. Why would we want to do that?

Reply:
I see what you're trying to accomplish now. When "logBackoffEnabled" is specified on construction, the logger should log its first batch fast then progressively back off, to effect more responsive startup time logging.